src/main/java/com/aws/greengrass/deployment/DeploymentService.java

```diff
@@ -504,18 +504,20 @@ private void cancelCurrentDeployment() {
                     .isCancellable()) {
                 logger.atInfo().log("Deployment already finished processing or cannot be cancelled");
             } else {
-                boolean canCancelDeploymentUpdateAction = context.get(UpdateSystemPolicyService.class)
-                        .discardPendingUpdateAction(((DefaultDeploymentTask) currentDeploymentTaskMetadata
-                                .getDeploymentTask()).getDeployment().getGreengrassDeploymentId());
-                if (!canCancelDeploymentUpdateAction) {
-                    logger.atWarn().kv(DEPLOYMENT_ID_LOG_KEY_NAME, currentDeploymentTaskMetadata.getDeploymentId())
-                            .kv(GG_DEPLOYMENT_ID_LOG_KEY_NAME,
-                                    currentDeploymentTaskMetadata.getGreengrassDeploymentId())
-                            .log("Cancelling deployment, changes may already have been applied. "
-                                    + "Make a new deployment to revert the update");
+                DeploymentTask deploymentTask = currentDeploymentTaskMetadata.getDeploymentTask();
```

Comment:
Are we adding this additional check?
For `KernelDeploymentTask` tasks, `isCancellable` is false.

Reply:
> Why are we adding this additional check?

We are adding this check because only `DefaultDeploymentTask` allows you to get the Deployment object, `KernelDeploymentTask` does not have a getter for it.

> For KernelDeploymentTask tasks, isn't isCancellable false?

We want to make `KernelDeploymentTask` cancellable as part of this code change